pkg/messageFormat/ast.go

```diff
@@ -0,0 +1,101 @@
+package messageFormat
+
+import (
+	"fmt"
+	"strings"
+)
+
+type NodeMatch struct {
+	Selectors []NodeExpr
+	Variants  []NodeVariant
+}
+
+type NodeVariable struct {
+	Name string
+}
+
+type NodeLiteral struct {
+	Value string
+}
+
+type NodeText struct {
+	Text string
+}
+
+type NodeExpr struct {
+	Value    interface{} // NodeVariable | NodeLiteral
+	Function NodeFunction
+}
+
+type NodeVariant struct {
+	Keys    []string
+	Message []interface{}
+}
+
+type NodeFunction struct {
+	Name string
+}
+
+func TokensToMessageFormat(tokens []Token) (NodeMatch, error) {
+	var (
+		nodeMatch           NodeMatch
+		currentNodeVariant  NodeVariant
+		currentNodeExpr     NodeExpr
+		isPlaceholderClosed bool
+		currentLevel        int
+		emptyToken          Token
+	)
+
+	// If message format starts with placeholder {
+	if tokens[0].Type == PlaceholderOpen {
+		tokens[0] = emptyToken
+		tokens[len(tokens)-1] = emptyToken
+	}
+
+	for i, token := range tokens {
+		switch token.Type { //nolint:exhaustive
```

Comment:
add a case with all the missing types to avoid the use of nolint:exhaustive
case x, y, z:
// noop